src/libcore/str/pattern.rs

```diff
@@ -715,16 +715,13 @@ impl<'a, 'b> Pattern<'a> for &'b str {
     /// Checks whether the pattern matches at the front of the haystack
     #[inline]
     fn is_prefix_of(self, haystack: &'a str) -> bool {
-        haystack.is_char_boundary(self.len()) &&
```

Comment:
This is interesting. According to `git blame`, @bluss added this in 2015. But yeah, this looks unnecessary to me and I agree with the change.